### PR TITLE
feat: add fetch api inside handleRegister function

### LINKED ISSUE #2: Register Interaction

### PR DESCRIPTION
**Pull request by ilmi**
I've add register interaction

### CODE REVIEW

File: src/app/register/components/RegisterForm.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        name,
+        email,
+        password,
+      }),
+    });
+    console.log(res.json());
```
Comment (alfonsusac) [inline src/app/register/components/RegisterForm.jsx]: Hi, please put this part into a separate function in `/lib/fetchAPI.js`.
The function is called `requestRegister()` accepting 3 parameters: name, email, and password.
It should return the json of the response.
Then you can call that `requestRegister()` in this compoennt.

Comment (alfonsusac) [inline src/app/register/components/RegisterForm.jsx]: res.json() needs to be awaited.

Make sure you have tested that this function works 👍

File: src/app/register/page.js
Comment (alfonsusac) [inline src/app/register/page.js:1]: lgtm

**Review by alfonsusac — CHANGES_REQUESTED** [review]

File: src/app/register/components/RegisterForm.jsx (hunk 1 above)
Comment (ilmi) [reply src/app/register/components/RegisterForm.jsx]: ok noted

Comment (ilmi) [reply src/app/register/components/RegisterForm.jsx]: ok noted

Comment (ilmi) [reply src/app/register/components/RegisterForm.jsx]: done, please kindly check

Comment (ilmi) [reply src/app/register/components/RegisterForm.jsx]: done

**Review by alfonsusac — COMMENTED** [review]
LGTM

File: src/lib/fetchAPI.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      email,
+      password,
+    }),
+  });
+
+  const data = await res.json();
+  return data;
+}
+
+async function requestRegister(name, email, password) {
```
Comment (alfonsusac) [inline src/lib/fetchAPI.js]: Convert this to `export async function`

File: src/lib/fetchAPI.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
+async function requestLogin(email, password) {
```
Comment (alfonsusac) [inline src/lib/fetchAPI.js]: Convert this to `export async function`

File: src/lib/fetchAPI.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      email,
+      password,
+    }),
+  });
+
+  const data = await res.json();
+  return data;
+}
+
+export { requestLogin, requestRegister };
```
Comment (alfonsusac) [inline src/lib/fetchAPI.js]: Remove these and convert to `export async function`

**Review by alfonsusac — CHANGES_REQUESTED** [review]
Sorry for the last minute changes. Only small minor adjustment needed.

File: src/lib/fetchAPI.js (hunk 2 above)
Comment (ilmi) [reply src/lib/fetchAPI.js]: done

File: src/lib/fetchAPI.js (hunk 4 above)
Comment (ilmi) [reply src/lib/fetchAPI.js]: done

File: src/lib/fetchAPI.js (hunk 3 above)
Comment (ilmi) [reply src/lib/fetchAPI.js]: done

**Review by alfonsusac — APPROVED** [review]

**Comment by alfonsusac** [comment]
Approved @ilmi please resolve the conflict

**Pull request merged**